pkg/cmd/triggertemplate/describe.go

```diff
@@ -62,10 +61,21 @@ const describeTemplate = `{{decorate "bold" "Name"}}:	{{ .TriggerTemplate.Name }
 {{- end }}
 
 {{decorate "resources" ""}}{{decorate "underline bold" "ResourceTemplates\n"}}
-
-{{- $value := getResourceTemplate .TriggerTemplate }}
-{{-  if ne $value "" }}
-{{ $value }}
+ NAME	GENERATENAME	KIND	APIVERSION
+{{- range $p := .TriggerTemplate.Spec.ResourceTemplates }}
+{{- $value := getResourceTemplate $p }}
+{{- if not $value }}
+{{- else }}
+{{- if eq $value.GetName "" }}
+ {{decorate "bullet" "---" }}	{{ $value.GetGenerateName }}	{{ $value.GetKind }}	{{ $value.GetAPIVersion }}
```

Comment:
you can simply return `---` from the function if empty and can remove all `if` and `else if` logic here

Reply:
all values are not empty
there may be possibilities like name can be empty or generateName can be empty so to handle those things added conditions

Reply:
I mean this, here in the template you can just write

```
{{decorate "resources" ""}}{{decorate "underline bold" "ResourceTemplates\n"}}
{{-  if ne $value "" }}	 NAME	GENERATENAME	KIND	APIVERSION
{{ $value }}	{{- range $p := .TriggerTemplate.Spec.ResourceTemplates }}
{{- $value := getResourceTemplate $p }}
 {{decorate "bullet" formatName $value.GetName }}	{{ formatName $value.GetGenerateName }}	{{ $value.GetKind }}	{{ $value.GetAPIVersion }}
```

And can have a function like
```
fund formatName(name string) {
    if name == nil {
        return "---"
    }
    return name
}
```